Modules/_abc.c

```diff
@@ -39,50 +39,24 @@ typedef struct {
 static void
 abc_data_dealloc(_abc_data *self)
 {
-    Py_DECREF(self->_abc_registry);
-    Py_DECREF(self->_abc_cache);
-    Py_DECREF(self->_abc_negative_cache);
+    Py_XDECREF(self->_abc_registry);
+    Py_XDECREF(self->_abc_cache);
+    Py_XDECREF(self->_abc_negative_cache);
     Py_DECREF(self->_abc_negative_cache_version);
     Py_TYPE(self)->tp_free(self);
 }
 
 static PyObject *
 abc_data_new(PyTypeObject *type, PyObject *args, PyObject *kwds)
 {
-    PyObject *registry = NULL, *cache = NULL, *negative_cache = NULL;
-    registry = PySet_New(NULL);
-    if (registry == NULL) {
-        return NULL;
-    }
-    cache = PySet_New(NULL);
-    if (cache == NULL) {
-        Py_DECREF(registry);
-        return NULL;
-    }
-    negative_cache = PySet_New(NULL);
-    if (negative_cache == NULL) {
-        Py_DECREF(registry);
-        Py_DECREF(cache);
-        return NULL;
-    }
-
     _abc_data *self = (_abc_data *) type->tp_alloc(type, 0);
     if (self == NULL) {
-        goto error;
+        return NULL;
     }
 
-    self->_abc_registry = registry;
-    self->_abc_cache = cache;
-    self->_abc_negative_cache = negative_cache;
```

Comment:
Do we want to explicitly initialize them to NULLs? `tp_alloc` might not be using calloc.